### PR TITLE
Fix check of whether popover update message contains a header

### LINKED ISSUE #1011: Popover title isn't getting removed after being added
Do the following in the following to repro the issue:

1. Click "Show popover"
2. Click "Hide popover"
3. Click "Show popover"
4. Switch on "Add a popover title"
5. Switch off "Add a popover title"

Step 5 should remove the title from popover, but it doesn't. This issue was surfaced after a recent fix in #747 by an existing popover test https://rstudio.github.io/shinycoreci/results/2024/03/15/#316-bslib-popovers-macOS-4.3

```r
library(shiny)
library(bslib)
library(plotly)

ui <- page_sidebar(
  title = "Popover tests",
  card(
    card_header(
        popover(
          span(
            "Card title with popover",
            bsicons::bs_icon("question-circle-fill")
          ),
          "Popover message",
          id = "popover",
          placement = "right"
        )
    ),
    plotlyOutput("bars")
  ),
  sidebar = list(
    textInput("popover_msg", "Enter a popover message", "Popover message"),
    actionButton("show_popover", "Show popover", class = "mb-3"),
    actionButton("hide_popover", "Hide popover"),
    br(),
    input_switch("show_title", "Add a popover title"),
    conditionalPanel(
      "input.show_title",
      textInput("popover_title", "Enter a title", "Popover title"),
    )
  )
)

server <- function(input, output, session) {
  observe({
    update_popover(
      "popover", input$popover_msg,
      title = if (input$show_title) input$popover_title
    )
  })

  observeEvent(input$show_popover, {
    toggle_popover("popover", show = TRUE)
  })

  observeEvent(input$hide_popover, {
    toggle_popover("popover", show = FALSE)
  })

  output$bars <- renderPlotly({
    plot_ly(diamonds, x = ~cut)
  })
}

shinyApp(ui, server)
```

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #1011

### CODE REVIEW

File: srcts/src/components/webcomponents/popover.ts [hunk 1]
```diff
@@ -385,7 +385,7 @@ export class BslibPopover extends BslibElement {
       trigger: this.triggerElement,
       content: {
         // eslint-disable-next-line @typescript-eslint/naming-convention
-        ".popover-header": hasHeader(newHeader) ? newHeader : "",
+        ".popover-header": header && header.html ? newHeader : "",
```
Comment (gadenbuie) [inline srcts/src/components/webcomponents/popover.ts:388]: This is one of those cases where I think an intermediate variable would be helpful just from a readability perspective.

e.g. something like this

```suggestion
        ".popover-header": has_updated_header ? newHeader : "",
```

where

```js
has_updated_header: header && header.html
```

comes very early on, maybe even just after `header` is destructured from the message data.

That said, I'm confused by this comment above with `getOrCreateElement()`. 

https://github.com/rstudio/bslib/blob/3941bc7ee7985e8fc12a3d39d664900758bdf045/srcts/src/components/webcomponents/popover.ts#L357-L361

I guess I'm confused why `header` goes through `getOrCreateElement()` if we end up ignoring its value when `header.html` exists?

Comment (gadenbuie) [reply srcts/src/components/webcomponents/popover.ts:388]: One other note, I'm guessing this behavior was introduced by changes in #747, but I'm having a hard time seeing how the bug was introduced or how this fixes it.

Normally, for a one-line change for a bug like this, I'd expect the fix to feel a little more obvious. But this feels _subtle_. I'm not sure you need to do anything with that information, other than maybe to explain why this is the fix in this PR.

Comment (cpsievert) [reply srcts/src/components/webcomponents/popover.ts]: Thanks for the great observations and apologies for throwing up this half-baked PR late on a Friday evening (wanted to jot my thoughts down, but should've left as a draft). 

After coming back to this, I'm embarrassed to say the proposed fix here is way off-basis: `update_popover(header = NULL)` shouldn't remove the header, only `update_popover(header = character(0))` should. The testing app that surfaced this issue should be passing `character(0)` but is currently passing `NULL`, and it was only ever working before due to a weird interaction with #747.

Comment (gadenbuie) [reply srcts/src/components/webcomponents/popover.ts]: No worries, thanks for the update!

**Pull request closed without merging**